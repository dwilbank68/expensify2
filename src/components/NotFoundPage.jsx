import React, {Component} from 'react';
import {Link} from 'react-router-dom';
// import PropTypes from 'prop-types';

// import EditExpense from './EditExpense.jsx';
class EditExpense extends Component {

    // constructor(props, context){
    //     super(props, context);
    //     this.state = {
    //         whatever:{}
    //     }
    //    this.handleClick = this.handleClick.bind(this)
    // }


    // handleClick(e) {
    //    
    //    this.setState(prevState => {
    //        return {}
    //    })
    // }

    /////////// ALTERNATIVE 1 - if using create-react-app
    // state = { whatever: false };

    /////////// ALTERNATIVE 2 - if using ES2016 property initializer
    // no more constructor or 'this' binding required
    //
    // state = { whateve': false }
    //
    // handleClick = (e) => {
    //    this.setState(prevState => {
    //        return {}
    //    })
    // }


    render() {
        return (
            <header className="header">
                404 - <Link to="/">Go Home</Link>
            </header>
        );
    }
}

// EditExpense.defaultProps = {};
// EditExpense.propTypes = {
//     name:        PropTypes.string.isRequired,
//     id:          PropTypes.oneOfType([ PropTypes.number, PropTypes.string ]).isRequired,
//     message:     PropTypes.shape({ title: PropTypes.string, text: PropTypes.string }).isRequired,
//     comments:    PropTypes.arrayOf(React.PropTypes.object),
//     date:        PropTypes.instanceOf(Date)
// };
//
// PropTypes -> array, bool, func, number, object, string, symbol

// EditExpense.contextTypes = {
//     router: React.PropTypes.object.isRequired
// }
// (lets you do 'this.context.router.push('/wherever');

export default EditExpense;

// remember to use 'this' binding now (choose one, #1 is best)
// 1. In constructor (see constructor above)
// 2. .bind(this) in your render ( onClick={this.onClick.bind(this)} )
// 3. Arrow functions in your render ( onClick={() => {}} )
// 4. _.bindAll(this,'methodName','...')

// }