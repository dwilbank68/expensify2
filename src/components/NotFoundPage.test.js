import React from 'react';
import {shallow} from 'enzyme';

import NotFoundPage from './NotFoundPage.jsx';

import '../setupTests.js';

test('should render NotFoundPage with expense', () => {
    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper).toMatchSnapshot();
})