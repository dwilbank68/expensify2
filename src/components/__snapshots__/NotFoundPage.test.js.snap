// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotFoundPage with expense 1`] = `
<header
  className="header"
>
  404 - 
  <Link
    replace={false}
    to="/"
  >
    Go Home
  </Link>
</header>
`;
